src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java

```diff
@@ -0,0 +1,448 @@
+		return cieProviders;
+	}
+
+	public List<ProviderInfo> getSpidProviders() {
+		return spidProviders;
+	}
+
+	public Map<String, String> getIdentityProviders(OIDCProfile profile) {
+		Map<String, String> result = new HashMap<>();
+
+		if (OIDCProfile.CIE.equals(profile)) {
+			for (ProviderInfo provider : cieProviders) {
+				result.put(provider.getSubject(), provider.getTrustAnchor());
+			}
+		}
+		else if (OIDCProfile.SPID.equals(profile)) {
+			for (ProviderInfo provider : spidProviders) {
+				result.put(provider.getSubject(), provider.getTrustAnchor());
+			}
+		}
+
+		return result;
+	}
+
+	public List<String> getTrustAnchors() {
+		return trustAnchors;
+	}
+
+	public RelyingParty getRelyingParty() {
+		return relyingParty;
+	}
+
+	public Hosts getHosts() {
+		return hosts;
+	}
+
+	public void setDefaultTrustAnchor(String defaultTrustAnchor) {
+		this.defaultTrustAnchor = defaultTrustAnchor;
+	}
+
+	public void setTrustAnchors(List<String> trustAnchors) {
+		this.trustAnchors = trustAnchors;
+	}
+
+	public JSONObject toJSON() {
+		JSONObject json = new JSONObject();
+
+		json.put("defaultTrustAnchor", defaultTrustAnchor);
+		json.put("trustAnchors", trustAnchors);
+		json.put("relyingParty", relyingParty.toJSON());
+		json.put("spidProviders", new JSONArray(spidProviders));
+		json.put("cieProviders", new JSONArray(cieProviders));
+		json.put("hosts", hosts.toJSON());
+
+		return json;
+	}
+
+	public OpenIdCredentialIssuer getOpenidCredentialIssuer() {
+		return openidCredentialIssuer;
+	}
+
+	public void setOpenidCredentialIssuer(OpenIdCredentialIssuer openidCredentialIssuer) {
+		this.openidCredentialIssuer = openidCredentialIssuer;
+	}
+
+	public FederationEntity getFederationEntity() {
+		return federationEntity;
+	}
+
+	public void setFederationEntity(FederationEntity federationEntity) {
+		this.federationEntity = federationEntity;
+	}
+
+
+	public String getFederationTrustChainUrl() {
+		return federationTrustChainUrl;
+	}
+
+	public void setFederationTrustChainUrl(String federationTrustChainUrl) {
+		this.federationTrustChainUrl = federationTrustChainUrl;
+	}
+
+
+	public static class Hosts extends BaseConfig {
+
+		public String getTrustAnchor() {
+			return trustAnchor;
+		}
+
+		public String getCieProvider() {
+			return cieProvider;
+		}
+
+		public String getRelyingParty() {
+			return relyingParty;
+		}
+
+		public void setTrustAnchor(String trustAnchor) {
+			this.trustAnchor = trustAnchor;
+		}
+
+		public void setCieProvider(String cieProvider) {
+			this.cieProvider = cieProvider;
+		}
+
+		public void setRelyingParty(String relyingParty) {
+			this.relyingParty = relyingParty;
+		}
+
+		public JSONObject toJSON() {
+			return new JSONObject()
+					.put("trust-anchor", trustAnchor)
+					.put("cie-provider", cieProvider)
+					.put("relying-party", relyingParty);
+		}
+
+		private String trustAnchor = "127.0.0.1";
+		private String cieProvider = "127.0.0.1";
+		private String relyingParty = "127.0.0.1";
```

Comment:
why using ipv4 private/local ips as constant in the code?

Reply:
Those are a default value, real params are configured via application.yml.